src/Microsoft.AspNet.OData.Shared/Query/ODataQueryOptions.cs

```diff
@@ -875,10 +875,18 @@ private string GetAutoSelectRawValue()
         private string GetAutoExpandRawValue()
         {
             var expandRawValue = RawValues.Expand;
-            IEdmEntityType baseEntityType = Context.TargetStructuredType as IEdmEntityType;
+            IEdmEntityType baseEntityType;
+            if (Context.ElementClrType != null)
+            {
+                IEdmType edmType = Context.Model.GetTypeMappingCache().GetEdmType(Context.ElementClrType, Context.Model)?.Definition;
+                baseEntityType = edmType as IEdmEntityType;
+            }
+            else
```

Comment:
Please add curly braces to this branch in an attempt to avoid maintainability mistakes